### PR TITLE
John conroy/rotate chart labels

### PR DESCRIPTION
**Pull request by john-conroy**
Towards #2593. There's some refactoring of common chart components/props that could be done, but I'd like to revisit. 

Also adds tooltips for the x axis tick labels.

Screen Shot 2022-09-20 at 4 32 06 PM
Screen Shot 2022-09-20 at 4 31 47 PM

### CODE REVIEW

File: context/app/static/js/shared-styles/charts/VerticalStackedBarChart/VerticalStackedBarChart.jsx [hunk 1]
```diff
@@ -43,8 +52,15 @@ function VerticalStackedBarChart({
...
-        <Group top={margin.top} left={margin.left}>
+        <GridRows
+          top={updatedMargin.top}
+          left={updatedMargin.left}
+          scale={yScale}
+          width={xWidth}
+          stroke="black"
+          opacity={0.2}
+        />
+        <Group top={updatedMargin.top} left={updatedMargin.left}>
```
Comment (mccalluc) [inline context/app/static/js/shared-styles/charts/VerticalStackedBarChart/VerticalStackedBarChart.jsx:63]: Maybe factor here and above out into a trivial component?
```
<GridRowsGroup  margin={updatedMargin} scale={yScale} width={xWidth}>
  ...
```

File: context/app/static/js/shared-styles/charts/utils.js [hunk 2]
```diff
@@ -5,4 +5,11 @@ function getChartDimensions(parentWidth, parentHeight, margin) {
   return { xWidth, yHeight };
 }
 
-export { getChartDimensions };
+function trimStringWithMiddleEllipsis(str) {
+  if (str.length > 25) {
+    return `${str.substr(0, 15)}...${str.substr(str.length - 10, str.length)}`;
+  }
+  return str;
```
Comment (mccalluc) [inline context/app/static/js/shared-styles/charts/utils.js:12]: Regex might be good here?

```suggestion
  return str.replace(/^(.{15}).+(.{10})$/,'$1...$2');
```

**Review by mccalluc — COMMENTED** [review]
Looks good. Some comments you can take or leave.

File: context/app/static/js/shared-styles/charts/VerticalStackedBarChart/VerticalStackedBarChart.jsx [hunk 3]
```diff
@@ -23,8 +24,16 @@ function VerticalStackedBarChart({
...
 }) {
-  const { xWidth, yHeight } = getChartDimensions(parentWidth, parentHeight, margin);
+  const tickLabelSize = 11;
+  const longestLabelSize = useLongestLabelSize({
+    labels: xAxisTickLabels.map((label) => trimStringWithMiddleEllipsis(label)),
+    labelFontSize: tickLabelSize,
+  });
+  const updatedMargin = { ...margin, bottom: Math.max(margin.bottom, longestLabelSize + 40) };
+
+  const { xWidth, yHeight } = getChartDimensions(parentWidth, parentHeight, updatedMargin);
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/shared-styles/charts/VerticalStackedBarChart/VerticalStackedBarChart.jsx]: Pulled all of the hooks to a central hook.

File: context/app/static/js/shared-styles/charts/VerticalStackedBarChart/VerticalStackedBarChart.jsx (hunk 1 above)
Comment (john-conroy) [reply context/app/static/js/shared-styles/charts/VerticalStackedBarChart/VerticalStackedBarChart.jsx]: Done.

**Pull request merged**